components/slider-carousel/src/ReuseSliderCarousel.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useRef, useEffect } from "react";
+import { useSlider } from "@locoworks/reusejs-toolkit-react-hooks";
+import { twMerge } from "tailwind-merge";
+
+interface ReuseSliderCarouselInterface {
+  slideInterval: number;
+  slides: React.ReactNode[];
+  loop: boolean;
+  wrapperClasses?: string;
+  sliderContainerClasses?: string;
+  enableButtons?: boolean | false;
+  prefixButton?: React.ReactNode;
+  suffixButton?: React.ReactNode;
+  buttonClasses?: string;
+  animationStyle?: string;
+}
+
+const ReuseSliderCarousel = ({
+  slideInterval,
+  slides,
+  loop,
+  wrapperClasses,
+  sliderContainerClasses,
+  enableButtons,
+  prefixButton,
+  suffixButton,
+  buttonClasses,
+  animationStyle,
+}: ReuseSliderCarouselInterface) => {
+  const {
+    currentSlide,
+    currentSlideIndex,
+    moveToNextSlide,
+    moveToPreviousSlide,
+    pauseSlider,
+    moveSlider,
+  } = useSlider({
+    slideInterval,
+    slides,
+    loop,
+  });
+
+  const parentRef = useRef<HTMLDivElement>(null);
+  let parent: any = parentRef.current;
+  const [scrollLeft, setScrollLeft] = useState<number>(0);
+  const [childWidth, setChildWidth] = useState<number>(0);
+
+  const HandleClickEvent = (e: any) => {
+    if (!enableButtons) {
+      const rect = e.currentTarget.getBoundingClientRect();
+      const clickPosition = e.clientX - rect.left;
+      const halfWidth = rect.width / 2;
+
+      if (clickPosition > halfWidth) {
+        moveToNextSlide();
+      } else {
+        moveToPreviousSlide();
+      }
+    }
+  };
+
+  const defaultWrapperClasses =
+    "relative flex items-center justify-center h-full w-full bg-gray-200";
+
+  const defaultSliderContainerClasses =
+    "flex justify-center items-center h-full w-full shrink-0";
+
+  const defaultButtonClasses =
+    "flex h-10 w-10 rounded-full text-center mx-6 bg-white items-center justify-center cursor-pointer";
+
+  const finalWrapperClasses = twMerge(defaultWrapperClasses, wrapperClasses);
+  const finalSliderContainerClasses = twMerge(
+    defaultSliderContainerClasses,
+    sliderContainerClasses
+  );
+
+  const finalButtonClasses = twMerge(defaultButtonClasses, buttonClasses);
+
+  const previousButton = (
+    <div className={finalButtonClasses} onClick={moveToPreviousSlide}>
+      {prefixButton ? prefixButton : <span>&#8810;</span>}
+    </div>
+  );
+
+  const nextButton = (
+    <div className={finalButtonClasses} onClick={moveToNextSlide}>
+      {suffixButton ? suffixButton : <span>&#8811;</span>}
+    </div>
+  );
+
+  const resumeSlides = () => {
```

Comment:
Can we extract this functionality without any inbuilt styles into a headless component?

Reply:
Not without UI Just without any styling, Would that be possible

Reply:
yes This is Possible but, user has to pass component with all the styling fro parent and child element.

Reply:
This is Implemented and added to same PR please review.